_posts/2022-10-19-quarkus-observability-roadmap-2023.adoc

```diff
@@ -0,0 +1,94 @@
+
+== OpenTracing Archived
+
+At the beginning of 2022, the https://https://www.cncf.io[CNCF] announced that it was
+https://www.cncf.io/blog/2022/01/31/cncf-archives-the-opentracing-project/[archiving the OpenTracing project], meaning
+that the project is no longer maintained. An expected decision since the merger between
+https://opentracing.io[OpenTracing] and http://opencensus.io/[OpenCensus] to form
+http://opentelemetry.io[OpenTelemetry].
+
+https://quarkus.io[Quarkus] released the first version of the OpenTelemetry Extension with Quarkus 1.13.0.Final, in Q1
+of 2021. The initial support was only for HTTP Tracing but slowly moved to other components, including CDI, gRPC,
+Reactive, Messaging, and many more, while also adopting and adjusting to all the new APIs released by OpenTelemetry
+Tracing. Our goal is to continue this work, expand OpenTelemetry Tracing to the remaining components and close the
+integration gaps between https://quarkus.io[Quarkus] and Tracing.
+
+*Moving forward, https://github.com/smallrye/smallrye-opentracing/[SmallRye OpenTracing] and all of the
+https://opentracing.io[OpenTracing] work in https://quarkus.io[Quarkus] are now deprecated!*
+
+While things will likely continue to work, we encourage all our
+users to move to http://opentelemetry.io[OpenTelemetry], for all the Tracing capabilities. In the following upcoming
+versions, we may start to remove https://opentracing.io[OpenTracing] from some extensions and continue that work until
+we completely remove https://opentracing.io[OpenTracing] from https://quarkus.io[Quarkus].
+
+For http://microprofile.io[MicroProfile] users out there that rely on
+https://github.com/eclipse/microprofile-opentracing[MicroProfile OpenTracing], MicroProfile 6.0, which will be
+supported by Quarkus 3 (and the Jakarta rename), ships
+https://github.com/eclipse/microprofile-telemetry[MicroProfile Telemetry Tracing]. This new specification defines the
+behaviors that allow http://microprofile.io[MicroProfile] applications to participate in an environment where
+distributed tracing is enabled via OpenTelemetry Tracing.
+
+== Metrics and Micrometer
+
+On the Metrics front, it became evident that while https://github.com/eclipse/microprofile-metrics[MicroProfile Metrics]
+and https://github.com/smallrye/smallrye-metrics/[SmallRye Metrics] were acceptable for simple
+use cases, they had some issues with more advanced ones. It also didn't help that users from different technologies had
+to learn new APIs and constructs to extract metrics information from their applications. We made a move to adopt
+https://micrometer.io[Micrometer] to resolve some of these issues while at the same time offering a less intrusive
+ways to monitor https://quarkus.io[Quarkus] applications.
+
+We intend to keep the https://micrometer.io[Micrometer] approach and expand it to the remaining areas. With
+OpenTelemetry Metrics reaching a stable milestone and https://micrometer.io[Micrometer] compromise to support
```

Comment:
It is not clear enough what we'll do.
Use Micrometer through OTel?
Use Micrometer and OTel Metrics?
Just Use Otel Metrics?

Reply:
I believe we still have to figure out that correctly, so I think it is acceptable to leave it open. If you and @ebullient already know how to proceed, feel free to propose a different statement.

Reply:
I'd prefer to use Micrometer through OTel and also provide OTel metrics support. However, we need to chat on creating an abstraction that simplifies implementing this approach.

Reply:
Do you think it is relevant to include that detail here? Or can you leave it open?

Reply:
This is important and should be discussed.

Reply:
Discussed on a meeting and agreed